integration/utils/blob_service.go

```diff
@@ -0,0 +1,94 @@
+package utils
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ipfs/go-blockservice"
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-datastore"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+
+	"github.com/onflow/flow-go/module/blobs"
+	"github.com/onflow/flow-go/module/component"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	"github.com/onflow/flow-go/network"
+)
+
+type testBlobService struct {
+	component.Component
+	blockService blockservice.BlockService
+	blockStore   blockstore.Blockstore
+}
+
+var _ network.BlobService = (*testBlobService)(nil)
+var _ component.Component = (*testBlobService)(nil)
+
+// WithHashOnRead sets whether or not the blobstore will rehash the blob data on read
+// When set, calls to GetBlob will fail with an error if the hash of the data in storage does not
+// match its CID
+func WithHashOnRead(enabled bool) network.BlobServiceOption {
+	return func(bs network.BlobService) {
+		bs.(*testBlobService).blockStore.HashOnRead(enabled)
+	}
+}
+
+// NewLocalBlobService creates a new BlobService that only interacts with its local datastore
+func NewLocalBlobService(
+	ds datastore.Batching,
+	opts ...network.BlobServiceOption,
+) *testBlobService {
+	bs := &testBlobService{
+		blockStore: blockstore.NewBlockstore(ds),
+	}
+
+	for _, opt := range opts {
+		opt(bs)
+	}
+
+	cm := component.NewComponentManagerBuilder().
+		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			bs.blockService = blockservice.New(bs.blockStore, nil)
```

Comment:
A side note: This now causes panic on `Close()` below given that `nil` as passed as a second argument: 
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x12fa7fe]
goroutine 2929 [running]:
github.com/ipfs/go-blockservice.(*blockService).Close(0xc003494ed0)
	/home/runner/go/pkg/mod/github.com/ipfs/go-blockservice@v0.2.0/blockservice.go:345 +0x7e
github.com/onflow/flow-go/integration/utils.NewLocalBlobService.func1({0x2277ed8, 0xc0033dec78}, 0xc00331bfc8)
	/home/runner/work/flow-go/flow-go/integration/utils/blob_service.go:57 +0x9c
github.com/onflow/flow-go/module/component.(*ComponentManager).Start.func2()
	/home/runner/work/flow-go/flow-go/module/component/component.go:248 +0xf0
created by github.com/onflow/flow-go/module/component.(*ComponentManager).Start
	/home/runner/work/flow-go/flow-go/module/component/component.go:245 +0x2de
FAIL	github.com/onflow/flow-go/integration/tests/access	81.203s
```

Reply:
thanks for pointing this out. I'll add a fix. Looks like calling `Close()` is unnecessary since all it does it call close on exchange (the second argument).